inter-page/support-bot/app.js

```diff
@@ -0,0 +1,152 @@
+ *
+ *
+ * Starter Project for Support bot of the Messenger Platform Inter-bot Tutorial
+ *
+ * Use this project as the starting point for following the tutorial.
+ *
+ * TODO: Waiting for link
+ *
+ */
+
+'use strict';
+
+// Imports dependencies and set up http server
+const
+    request = require('request'),
+    express = require('express'),
+    body_parser = require('body-parser'),
+    app = express().use(body_parser.json()); // creates express http server
+
+// Sets server port and logs message on success
+app.listen(process.env.PORT || 5001, () => console.log('webhook is listening'));
+
+app.post('/webhook', (req, res) => {
+
+    // Parse the request body from the POST
+    let body = req.body;
+
+    // Check the webhook event is from a Page subscription
+    if (body.object === 'page') {
+
+        body.entry.forEach(function (entry) {
+
+            // Gets the body of the webhook event
+            let webhook_event = entry.messaging[0];
+            console.log(webhook_event);
+
+            // Get the sender PSID
+            let sender_psid = webhook_event.sender.id;
+            console.log('Sender PSID: ' + sender_psid);
+
+            // Check if the event is a message or postback and
+            // pass the event to the appropriate handler function
+            if (webhook_event.message) {
+                handleMessage(sender_psid, webhook_event.message);
+            } else if (webhook_event.postback) {
+                handleIncoming(sender_psid, webhook_event.postback);
+            } else if (webhook_event.referral) {
+                handleIncoming(sender_psid, webhook_event.referral);
+            }
+        });
+
+        // Return a '200 OK' response to all events
+        res.status(200).send('EVENT_RECEIVED');
+
+    } else {
+        // Return a '404 Not Found' if event is not from a page subscription
+        res.sendStatus(404);
+    }
+
+});
+
+// Accepts GET requests at the /webhook endpoint
+app.get('/webhook', (req, res) => {
+
+    const VERIFY_TOKEN = process.env.TOKEN;
+
+    // Parse params from the webhook verification request
+    let mode = req.query['hub.mode'];
+    let token = req.query['hub.verify_token'];
+    let challenge = req.query['hub.challenge'];
+
+    // Check if a token and mode were sent
+    if (mode && token) {
+
+        // Check the mode and token sent are correct
+        if (mode === 'subscribe' && token === VERIFY_TOKEN) {
+
+            // Respond with 200 OK and challenge token from the request
+            console.log('WEBHOOK_VERIFIED');
+            res.status(200).send(challenge);
+
+        } else {
+            // Responds with '403 Forbidden' if verify tokens do not match
+            res.sendStatus(403);
+        }
+    }
+});
+
+
+function handleMessage(sender_psid, received_message) {
+    let response;
+
+    if (received_message.text) {
+        switch (received_message.text.replace(/[^\w\s]/gi, '').trim().toLowerCase()) {
```

Comment:
is this switch statement needed? Looks like it only has a default

Reply:
This was more so people could add to it in the future, stay or go?

Reply:
@ChrisChinchilla I'd say take it out